tests/multi_net/udp_data.py

```diff
@@ -3,34 +3,69 @@
 import socket
 
 NUM_NEW_SOCKETS = 4
-NUM_TRANSFERS = 4
+NUM_TRANSFERS = 10
+TOTAL_PACKETS = NUM_NEW_SOCKETS * NUM_TRANSFERS
+# If more than 75% of packets are lost, the test fails.
+PACKET_LOSS_THRESH = 0.75 * TOTAL_PACKETS
 PORT = 8000
 
 # Server
 def instance0():
+    seq = 0
     multitest.globals(IP=multitest.get_network_ip())
     multitest.next()
     for i in range(NUM_NEW_SOCKETS):
         s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
         s.bind(socket.getaddrinfo("0.0.0.0", PORT)[0][-1])
+        s.settimeout(0.250)
         multitest.broadcast("server ready")
         for j in range(NUM_TRANSFERS):
-            data, addr = s.recvfrom(1000)
-            print(data)
-            s.sendto(b"server to client %d %d" % (i, j), addr)
+            try:
+                data, addr = s.recvfrom(1000)
+            except:
+                continue
+            if int(data) == seq:
+                if seq < (TOTAL_PACKETS - PACKET_LOSS_THRESH):
+                    print(seq)
+                seq += 1
+                s.sendto(b"%d" % (seq), addr)
         s.close()
 
+    if (TOTAL_PACKETS - seq) > PACKET_LOSS_THRESH:
+        print(
+            "packet loss %.1f%% %d/%d"
+            % (((TOTAL_PACKETS - seq) / TOTAL_PACKETS * 100), seq, TOTAL_PACKETS)
+        )
+    else:
+        print("pass")
```

Comment:
This bit of code seems common to instance0 and instance1, so best if it's made into a function they can both call.

Reply:
Fixed

Reply:
looks like this changed was not pushed here??

Reply:
It did, I moved the code that was here to `loss_percent()`. What else do you want to make common ?

Reply:
I thought these 4 lines (the whole if-else) can be made a function, with argument `seq`.

Reply:
eg `print_stats(seq)`

Reply:
Okay fixed.